lib/kafkalib/consumer.go

```diff
@@ -10,3 +12,31 @@ type Consumer interface {
 	ReadMessage(ctx context.Context) (kafka.Message, error)
 	CommitMessages(ctx context.Context, msgs ...kafka.Message) error
 }
+
+type Reader struct {
+	*kafka.Reader
+	config kafka.ReaderConfig
+}
+
+func NewReader(config kafka.ReaderConfig) *Reader {
+	return &Reader{
+		Reader: kafka.NewReader(config),
+		config: config,
+	}
+}
+
+func ShouldReload(err error) bool {
+	if err == nil {
```

Comment:
This could go away, `errors.Is` will return `false` for `nil`.

Reply:
https://go.dev/play/p/547vdvNWgJQ

Reply:
ill do that tomorrow and also add a test